### PR TITLE
Improve prettier / eslint configuration

### LINKED ISSUE #2908: Improve prettier / eslint configuration
I've started to have a look at the lint+format configuration on both JHipster lite generator & generated apps, it seems there are some misconfigurations:
- when committing files, we only apply prettier but not eslint (for instance a `let` instead of a `const`)
- when editing a front-end file (with IntelliJ for instance), eslint does not detect (and fix) prettier issues
- useless configuration parts

I'll start to do a PR to show all this. I'm not a total expert and this is a lot of configuration, so I hope I will improve it and not break anything :stuck_out_tongue:

### PR DESCRIPTION
**Pull request by hdurix**
Fix #2908

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -2,13 +2,14 @@ module.exports = {
   env: {
     node: true,
   },
-  extends: ['eslint:recommended', 'plugin:vue/vue3-recommended', '@vue/eslint-config-typescript', 'prettier'],
+  extends: ['eslint:recommended', 'plugin:vue/vue3-recommended', '@vue/eslint-config-typescript', '@vue/prettier'],
```
Comment (hdurix) [inline .eslintrc.js:5]: `prettier` does not seem to hint prettier errors in ts files in IDEs (at least IntelliJ)

File: .eslintrc.js [hunk 2]
```diff
@@ -2,13 +2,14 @@ module.exports = {
...
   parserOptions: {
     parser: '@typescript-eslint/parser',
   },
   rules: {
     quotes: ['error', 'single', { avoidEscape: true }],
     '@typescript-eslint/no-unused-vars': ['error'],
     'vue/multi-word-component-names': 'off',
+    'prettier/prettier': ['error'],
```
Comment (hdurix) [inline .eslintrc.js:13]: to show prettier's errors as errors in IDEs

File: .lintstagedrc.js [hunk 3]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
-  '{,src/**/,webpack/}*.{md,json,yml,html,js,ts,tsx,css,scss,vue,java,xml}': ['prettier --write'],
+  '{,src/**/}*.{js,ts,tsx,vue}': ['eslint --fix'],
```
Comment (hdurix) [inline .lintstagedrc.js:2]: - eslint will do both lint + format (with calling prettier)
- webpack is not used

File: .prettierrc [hunk 4]
```diff
@@ -11,15 +11,5 @@ arrowParens: avoid
 # jsx and tsx rules:
 bracketSameLine: false
 
-# search additional plugin, following prettier 2.6.0
-pluginSearchDirs:
```
Comment (hdurix) [inline .prettierrc:15]: this is the default configuration, not sure this is still necessary (I checked and noticed nothing wrong, was able to format XML files)

File: .prettierrc [hunk 5]
```diff
@@ -11,15 +11,5 @@ arrowParens: avoid
...
-  - .
-
-# java rules:
-overrides:
```
Comment (hdurix) [inline .prettierrc:19]: as we only overrides with same values, not sure this is useful

File: src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java [hunk 6]
```diff
@@ -36,9 +36,9 @@ public JHipsterModule buildVueModule(JHipsterModuleProperties properties) {
         .addDevDependency(packageName("@typescript-eslint/parser"), VUE)
         .addDevDependency(packageName("@vitejs/plugin-vue"), VUE)
         .addDevDependency(packageName("@vue/eslint-config-typescript"), VUE)
+        .addDevDependency(packageName("@vue/eslint-config-prettier"), VUE)
```
Comment (hdurix) [inline src/main/java/tech/jhipster/lite/generator/client/vue/core/domain/VueModulesFactory.java:39]: same changes for generated apps

File: .eslintrc.js (hunk 1 above)
Comment (hdurix) [reply .eslintrc.js:5]: tried in vscode, I think this was not working either

File: package.json [hunk 7]
```diff
@@ -45,11 +45,11 @@
     "@types/sinon": "10.0.13",
     "@typescript-eslint/parser": "5.32.0",
     "@vitejs/plugin-vue": "3.0.1",
+    "@vue/eslint-config-prettier": "^7.0.0",
```
[the remark this answers is not in the document]
Comment (DanielFran) [reply package.json:48]: @hdurix We use normally fixed version by default

File: src/main/resources/generator/dependencies/vue/package.json [hunk 8]
```diff
@@ -17,10 +17,10 @@
     "@types/sinon": "10.0.13",
     "@typescript-eslint/parser": "5.32.0",
     "@vitejs/plugin-vue": "3.0.1",
+    "@vue/eslint-config-prettier": "^7.0.0",
```
Comment (DanielFran) [inline src/main/resources/generator/dependencies/vue/package.json:20]: @hdurix We use normally fixed version by default

**Comment by pascalgrimaud** [comment]
Thanks @hdurix : just need to check if there is no conflict between the eslint config and prettier config

Plz can you use the fixed version and we could merge it

File: package.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (hdurix) [reply package.json]: good catch, changed!

File: src/main/resources/generator/dependencies/vue/package.json (hunk 8 above)
Comment (hdurix) [reply src/main/resources/generator/dependencies/vue/package.json]: changed in latest version

**Review by pascalgrimaud — APPROVED** [review]

**Pull request merged**